File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"

#include<stdlib.h>
#include<string.h>

char *to_rna(const char *dna)
{
    char *rna_result = malloc(sizeof(char) * (1 + strlen(dna)));
    char *rna = rna_result;
    for (; *dna != '\0'; dna++, rna++) {
        switch (*dna) {
            case 'G':
                *rna = 'C';
                break;
            case 'C':
                *rna = 'G';
                break;
            case 'T':
                *rna = 'A';
                break;
            case 'A':
                *rna = 'U';
                break;
         }
    }
    *rna = '\0';
    return rna_result;
}

File: solutions/ocaml/rna-transcription/1/rna_transcription.ml
type dna = [ `A | `C | `G | `T ]
type rna = [ `A | `C | `G | `U ]

let to_rna strand =
    let convert (nuc: dna): rna =
      match nuc with
      | `A -> `U 
      | `T -> `A 
      | `C -> `G
      | `G -> `C
    in
      List.map convert strand